experimental/web_dashboard/lib/main.dart

```diff
@@ -0,0 +1,204 @@
+class AppShell extends StatefulWidget {
+  final String routeName;
+
+  AppShell(this.routeName);
+
+  @override
+  _AppShellState createState() => _AppShellState();
+}
+
+class _AppShellState extends State<AppShell> {
+  int _currentIndex = 0;
+
+  void initState() {
+   super.initState();
+   _currentIndex = _indexForRoute(widget.routeName);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AdaptiveScaffold(
+      title: Text('Dashboard App'),
+      body: Navigator(
+        key: _navigatorKey,
+        initialRoute: widget.routeName,
+        onGenerateRoute: (settings) {
+          var routeName = settings.name;
+          var pageWidget = _pageForRoute(routeName);
+
+          // Change the _currentIndex to highlight the correct menu item /
+          // BottomNavigationBarItem
+          //
+          // Use a post-frame callback to avoid calling setState() during the
+          // initial build phase. Right now this is being called in the wrong
+          // order. (Refreshing the browser with the initial route set to
+          // '/settings' invokes onGenerateRoute with '/' instead of
+          // '/settings'.
+          SchedulerBinding.instance.addPostFrameCallback((dur) {
+            setState(() {
+              _currentIndex = _indexForRoute(routeName);
+            });
+          });
+          // Instead of a SchedulerBinding, this check should ensure that
+          // setState() isn't called during the initial build phase
+          // (_currentIndex is set to the correct value in initState()).
+          //
+          // if (_currentIndex != _indexForRoute(routeName)) {
+          //   setState(() {
+          //      _currentIndex = _indexForRoute(routeName);
+          //   });
+          // }
+
+          return PageRouteBuilder(
+            pageBuilder: (context, animation, secondaryAnimation) {
+              return pageWidget;
+            },
+            transitionsBuilder:
+                (context, animation, secondaryAnimation, child) {
+              var curveTween = CurveTween(curve: Curves.easeInQuart);
+              return FadeTransition(
+                opacity: animation.drive(curveTween),
+                child: child,
+              );
+            },
+            settings: settings,
+          );
+        },
+      ),
+      currentIndex: _currentIndex,
+      destinations: [
+        AdaptiveScaffoldDestination(title: 'Dashboard', icon: Icons.dashboard),
+        AdaptiveScaffoldDestination(title: 'Metrics', icon: Icons.show_chart),
+        AdaptiveScaffoldDestination(title: 'Settings', icon: Icons.settings),
+      ],
+      onNavigationIndexChange: (idx) {
+        setState(() {
+          _currentIndex = idx;
+        });
+
+        // Use the inner Navigator to change the route.
+        var navState = _navigatorKey.currentState as NavigatorState;
+        navState.pushNamed(_routeForIndex(idx));
+      },
+      floatingActionButton: FloatingActionButton(
```

Comment:
I can imagine wanting this button to have different effects based on which page is loaded into the navigator inside the AdaptiveScaffold. Is that something that would be supported?

Reply:
I don't think the AdaptiveScaffold would cover that case. The user would be responsible for applying any effects (or hiding the FAB altogether) based on the `currentIndex`.